liquid_node/commands.py

```diff
@@ -211,6 +211,66 @@ def check_system_config():
         'the "vm.max_map_count" kernel parameter is too low, check readme'
 
 
+def start_job(job, hcl):
+    log.info('Starting %s...', job)
+    spec = nomad.parse(hcl)
+    nomad.run(spec)
+    job_checks = {}
+    for service, checks in nomad.get_health_checks(spec):
+        if not checks:
+            log.warn(f'service {service} has no health checks')
+            continue
+        job_checks[service] = checks
+    return job_checks
+
+
+def populate_secrets(vault_secret_keys, core_auth_apps, liquid_job):
+    """Sets a secrets for every path given, then starts the liquid-core and creates OAuth2 entries for
+    every app. If self-hosted continuous integration is enabled, then the client id and secret are
+    added to vault
+    :vault_secrets_keys: A list of paths for which a secret is needed.
+    :core_auth_apps: A list of dictionary which contains app's name, vault_path and callback to create
+    Oauth2 entries
+    :liquid_job: The template for the "core job", which is the first to start
+    """
+
+    for path in vault_secret_keys:
+        ensure_secret_key(path)
+
+    if config.ci_enabled:
+        vault.set('liquid/ci/drone.github', {
+            'client_id': config.ci_github_client_id,
+            'client_secret': config.ci_github_client_secret,
+            'user_filter': config.ci_github_user_filter,
+        })
+        vault.set('liquid/ci/drone.docker', {
+            'username': config.ci_docker_username,
+            'password': config.ci_docker_password,
+        })
+
+    ensure_secret('liquid/rocketchat/adminuser', lambda: {
+        'username': 'rocketchatadmin',
+        'pass': random_secret(64),
+    })
+
+    for name in config.ALL_APPS:
+        ensure_secret(f'liquid/{name}/cookie', lambda: {
```

Comment:
Why do we need a separate key structure for these? Let's use the hardcoded list in `vault_secret_keys` (maybe make it a global?) -- we need to have these in the same place, more `for app in apps: write out this new chain of secrets`  aren't helping the readability of this code...

This also means you'll have to use `.secret_key` instead of `.secrets` in nomad templates.

Reply:
If we put those key in the vault_secret_keys, a 256 - bits secrets would be generated, which is not supported by the proxy (64bits is the current maximum at 29/09/2020)